include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -0,0 +1,487 @@
+    //!\hideinitializer
+    using reference = value_type;
+    //!\brief Use the const value type as reference type.
+    //!\hideinitializer
+    using const_reference = value_type const;
+    //!\brief Use the size_type of the underlying sequence.
+    //!\hideinitializer
+    using size_type = typename ranges::v3::size_type_t<inner_type>;
+    //!\brief Use the difference_type of the underlying sequence.
+    //!\hideinitializer
+    using difference_type = typename ranges::v3::difference_type_t<inner_type>;
+    //!\brief The iterator type of this container (a random access iterator).
+    //!\hideinitializer
+    using iterator = detail::random_access_iterator<gap_decorator_anchor_set>;
+    //!\hideinitializer
+    using const_iterator = detail::random_access_iterator<gap_decorator_anchor_set const>;
+    //!\}
+
+    /*!\name Constructors/Destructors
+     * \{
+     */
+    //!\brief Default constructor.
+    constexpr gap_decorator_anchor_set() = default;
+    //!\brief Copy constructor.
+    constexpr gap_decorator_anchor_set(gap_decorator_anchor_set const &) = default;
+    //!\brief Copy construction via assignment.
+    constexpr gap_decorator_anchor_set & operator=(gap_decorator_anchor_set const &) = default;
+    //!\brief Move constructor.
+    constexpr gap_decorator_anchor_set (gap_decorator_anchor_set && rhs) = default;
+    //!\brief Move assignment.
+    constexpr gap_decorator_anchor_set & operator=(gap_decorator_anchor_set && rhs) = default;
+    //!\brief Direct sequence assignment resets previously inserted gaps.
+    constexpr gap_decorator_anchor_set & operator=(inner_type const & sequence)
+    {
+        this->sequence = sequence;
+        anchors.clear();
+    };
+    //!\brief Use default deconstructor.
+    ~gap_decorator_anchor_set() = default;
+    //!\brief Construct by host and explicit position.
+    constexpr gap_decorator_anchor_set(inner_type const & sequence): sequence{&sequence} {};
+    //!\}
+
+    /*!\brief Returns the total length of the aligned sequence.
+     * \returns The total length of the aligned sequence.
+     *
+     * \par Complexity
+     *
+     * Constant.
+     *
+     * \par Exceptions
+     *
+     * No-throw guarantee.
+     */
+    size_type size() const noexcept
+    {
+        if (anchors.size())
+            return (*(anchors.rbegin())).second + sequence->size();
+        return sequence->size();
+    }
+
+    /*!\brief Insert a gap of length size at the aligned sequence iterator position.
+     * \returns A boolean flag indicating the success of the insertion operation.
+     *
+     * \par Complexity
+     *
+     * Average and worst case (insertion before last gap): o(k),
+     * Best case (back insertion): o(log k).
+     *
+     * \par Exceptions
+     *
+     * May throw std::set exceptions.
+     */
+    iterator insert_gap(iterator const it, size_type const size=1)
+    {
+        if (!size)
+            return it;
+        size_type const pos = it - this->begin();
```

Comment:
Yeah, that's more a habit to write explicitly `this->`, because I once ran into issues that the compiler could not find the function. But I omit now for all member calls.